packages/client/hmi-client/src/workflow/ops/decapodes/tera-decapodes-drilldown.vue

```diff
@@ -0,0 +1,209 @@
+
+const props = defineProps<{
+	node: WorkflowNode<DecapodesOperationState>;
+}>();
+
+const emit = defineEmits(['close', 'update-state']);
+const kernelManager = new KernelSessionManager();
+let editor: VAceEditorInstance['_editor'] | null;
+
+const modelId = computed(() => props.node?.inputs?.[0]?.value?.[0]);
+const notebookResponse = ref();
+const sampleAgentQuestions = ['Convert the model to equations please'];
+const contextLanguage = ref<string>('julia-1.10');
+const isRunningCode = ref<boolean>(false);
+const selectedOutputId = ref<string>(props.node.active ?? '');
+const outputs = computed(() => {
+	if (!isEmpty(props.node.outputs)) {
+		return [
+			{
+				label: 'Select outputs to display in operator',
+				items: props.node.outputs
+			}
+		];
+	}
+	return [];
+});
+
+const initializeEditor = (editorInstance: any) => {
+	editor = editorInstance;
+};
+const codeText = ref(
+	'# This environment contains the variable "model_config" to be read and updated'
+);
+
+const buildJupyterContext = () => ({
+	context: 'decapodes',
+	language: 'julia-1.10',
+	context_info: {
+		decapode_id: modelId.value ?? ''
+	}
+});
+
+const appendCode = (data: any, property: string) => {
+	codeText.value = codeText.value.concat(' \n', data.content[property] as string);
+	saveCodeToState(codeText.value);
+};
+
+const runFromCode = () => {
+	const code = editor?.getValue();
+	if (!code) return;
+
+	const messageContent = {
+		silent: false,
+		store_history: false,
+		code
+	};
+
+	isRunningCode.value = true;
+
+	kernelManager
+		.sendMessage('execute_request', messageContent)
+		.register('execute_input', (data) => {
+			console.log('execute_input', data.content);
+		})
+		.register('stream', (data) => {
+			console.log('stream', data.content);
+		})
+		.register('error', (data) => {
+			console.log('error', data.content.evalue);
+			logger.error(data.content.evalue);
+			isRunningCode.value = false;
+		});
+	// FIXME: There isnt really a proper response we receive for llm output for decapodes
+	// .register('decapodes_preview', (data) => {
+	// 	console.log('decapodes_preview', data.content);
+	// 	notebookResponse.value = data.content['application/json'];
+	// 	isRunningCode.value = false;
+	// });
+};
+
+const saveCodeToState = (code: string) => {
+	const state = cloneDeep(props.node.state);
+
+	// for now only save the last code executed, may want to save all code executed in the future
+	const codeHistoryLength = props.node.state.codeHistory.length;
+	const timestamp = Date.now();
+	if (codeHistoryLength > 0) {
+		state.codeHistory[0] = { code, timestamp };
+	} else {
+		state.codeHistory.push({ code, timestamp });
+	}
+
+	emit('update-state', state);
+};
+
+const inputChangeHandler = async () => {
+	if (!modelId.value) return;
+
+	const model = await getModel(modelId.value);
+
+	if (getModelType(model) !== AMRSchemaNames.DECAPODES) return;
+
+	codeText.value = props.node.state.codeHistory?.[0]?.code ?? '';
```

Comment:
Are we starting with blank code-text or your default code-text from above?

Reply:
it now defaults to the `defaultText`